File: android/app/src/main/java/com/demo/DashboardActivity.java
package com.demo;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import androidx.annotation.CallSuper;
import androidx.annotation.Nullable;

import com.facebook.react.ReactActivity;
import com.facebook.react.ReactActivityDelegate;
import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.WritableMap;
import com.facebook.react.modules.core.DeviceEventManagerModule;

/**
 * Activity to start from React Native JavaScript, triggered via
 */
public final class DashboardActivity extends ReactActivity {

    @Override
    @CallSuper
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_dashboard);

        Intent i = getIntent();
        Bundle bundle = i.getExtras();

        if (bundle != null) {
            String data = bundle.getString("DATA");
            TextView tvData = (TextView) findViewById(R.id.tvData);
            tvData.setText("DATA " + data);
        }

        findViewById(R.id.go_back_button1).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                onBackPressed();
            }
        });

        findViewById(R.id.go_back_button2).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {



                try {
                    getReactInstanceManager().getCurrentReactContext()
                            .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
                            .emit("DASHBOARD1", "eij eij ej ");

                } catch (Exception e) {
                    Log.e("ReactNative", "Caught Exception: " + e.getMessage());
                }

                onBackPressed();
            }
        });
    }

    @Override
    public void onBackPressed() {
        EditText etData = (EditText) findViewById(R.id.etSetData);

        try {
            getReactInstanceManager().getCurrentReactContext()
                    .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
                    .emit("DASHBOARD", etData.getText().toString());

        } catch (Exception e) {
            Log.e("ReactNative", "Caught Exception: " + e.getMessage());
        }
        super.onBackPressed();
    }
}

File: android/app/src/main/java/com/demo/DashboardActivity.kt
package com.demo

import android.os.Bundle
import android.util.Log
import android.view.View
import com.facebook.react.ReactActivity
import android.widget.TextView
import com.facebook.react.modules.core.DeviceEventManagerModule.RCTDeviceEventEmitter
import androidx.annotation.CallSuper
import androidx.viewbinding.ViewBinding
import com.demo.databinding.ActivityDashboardBinding
import com.demo.databinding.ActivitySigninBinding
import java.lang.Exception

/**
 * Activity to start from React Native JavaScript, triggered via
 */
class DashboardActivity : BaseActivity() {

    private lateinit var ui: ActivityDashboardBinding

    override fun layoutRes(): ViewBinding {
        return ActivityDashboardBinding.inflate(layoutInflater)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        ui = binding as ActivityDashboardBinding

        val i = intent
        val bundle = i.extras
        if (bundle != null) {
            val data = bundle.getString("DATA")

            ui.tvData.text = "DATA $data"
        }

        ui.goBackButton1.setOnClickListener {
            launchActivity(SignInActivity::class.java)
        }


        ui.goBackButton2.setOnClickListener {
            try {
                reactInstanceManager.currentReactContext
                    ?.getJSModule(RCTDeviceEventEmitter::class.java)
                    ?.emit("DASHBOARD1", "eij eij ej ")
            } catch (e: Exception) {
                Log.e("ReactNative", "Caught Exception: " + e.message)
            }
            onBackPressed()
        }

    }

}